src/read_write/codec.rs

```diff
@@ -140,3 +140,67 @@ where
         .unwrap()
         .mul_add(edge_length, min)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn plain_scalar() {
+        let value = 41.33333;
+        let min = 40.0;
+        let edge_length = 2.0;
+
+        let value_f32 = _encode::<f32>(value, min, edge_length);
+        let value_f32_decoded = decode(value_f32, min, edge_length);
+        assert!(
+            (value_f32_decoded - value).abs() < 0.0000001,
```

Comment:
nit: use `1e-7` instead of `0.0000001`? Similarly below.

Reply:
Done